### PR TITLE
docs: update "Getting started" guide

### PR DESCRIPTION
**Pull request by navendu-pottekkat**
Signed-off-by: Navendu Pottekkat <navendupottekkat@gmail.com>

Updates the "Getting started" page. Also updates the ACK installation. Once these changes are approved, it will be propagated to other installation docs as well. Also rearranges installation to the top.

### CODE REVIEW

**Comment by navendu-pottekkat** [comment]
@tao12345666333 @juzhiyuan Please review this PR.

**Comment by navendu-pottekkat** [comment]
No clue why the markdown link checks are failing.

File: docs/en/latest/getting-started.md [hunk 1]
```diff
@@ -21,68 +26,109 @@ title: Getting Started
...
-| `1.3.0` | `>= 2.7`, `2.10` is recommended. |
-| `1.2.0` | `>= 2.7`, `2.8` is recommended. |
-| `1.1.0` | `>= 2.7`, `2.7` is recommended. |
-| `1.1.0` | `>= 2.7`, `2.7` is recommended. |
-| `1.0.0` | `>= 2.7`, `2.7` is recommended. |
-| `0.6` | `>= 2.6`, `2.6` is recommended. |
-| `0.5` | `>= 2.4`, `2.5` is recommended. |
-| `0.4` |`>= 2.4`|
+| APISIX ingress controller | Recommended APISIX version |
+| ------------------------: | -------------------------: |
```
Comment (juzhiyuan) [inline docs/en/latest/getting-started.md:124]: Hello, after adding `Recommended` in L123, the meaning is changed in my opinion. 🤔 Like in L125, it means `>= 2.7` is recommended.

**Review by juzhiyuan — COMMENTED** [review]
Others LGTM

**Comment by juzhiyuan** [comment]
> No clue why the markdown link checks are failing.

@navendu-pottekkat Just rerun, the report says upstream link is 504 😅 but it works now.

File: docs/en/latest/getting-started.md (hunk 1 above)
Comment (navendu-pottekkat) [reply docs/en/latest/getting-started.md:124]: Yes. You are right. I will split this into two columns. I overlooked what is actually mentioned in the column.

**Comment by navendu-pottekkat** [comment]
> > No clue why the markdown link checks are failing.
> 
> @navendu-pottekkat Just rerun, the report says upstream link is 504 😅 but it works now.

@juzhiyuan I do not have access to re-run workflows!

File: docs/en/latest/getting-started.md (hunk 1 above)
Comment (navendu-pottekkat) [reply docs/en/latest/getting-started.md]: Updated.

**Review by tao12345666333 — APPROVED** [review]
LGTM

Thanks

File: docs/en/latest/getting-started.md [hunk 2]
```diff
@@ -21,68 +26,109 @@ title: Getting Started
...
+
+APISIX ingress controller can be configured using native Kubernetes ingress resources as well as a declarative and easy to use custom resources provided by APISIX. The APISIX ingress controller converts these resources to APISIX configuration.
+
+The examples below show how these differ. Both the examples configure a Route in APISIX that routes to an httpbin service as the Upstream.
+
+<Tabs
+groupId="resources"
+defaultValue="apisix"
+values={[
+{label: 'APISIX custom resource', value: 'apisix'},
```
Comment (lingsamuel) [inline docs/en/latest/getting-started.md:42]: Should be APISIX ingress custom resource.

**Review by lingsamuel — APPROVED** [review]

File: docs/en/latest/getting-started.md (hunk 2 above)
Comment (tao12345666333) [inline docs/en/latest/getting-started.md:42]:
```suggestion
{label: 'APISIX Ingress custom resource', value: 'apisix'},
```

**Comment by navendu-pottekkat** [comment]
These failing lint checks do not seem to be from this PR.

File: docs/en/latest/getting-started.md [hunk 3]
```diff
@@ -21,68 +26,109 @@ title: Getting Started
...
+            path: /
+            pathType: Prefix
+```
+
+</TabItem>
+</Tabs>
+
+APISIX ingress controller defines the CRDs [ApisixRoute](./concepts/apisix_route.md), [ApisixUpstream](./concepts/apisix_upstream.md), [ApisixTlx](concepts/apisix_tls.md), and [ApisixClusterConfig](concepts/apisix_cluster_config.md).
+
+APISIX also supports [service discovery](/docs/apisix/discovery/kubernetes/) through [Kubernetes service](https://kubernetes.io/docs/concepts/services-networking/service/) abstraction.
```
Comment (tao12345666333) [inline docs/en/latest/getting-started.md:95]: This link is wrong

File: docs/en/latest/getting-started.md [hunk 4]
```diff
@@ -21,68 +26,109 @@ title: Getting Started
...
+To summarize, APISIX ingress controller has the following features:
 
-## Installation on Prem
+- Declarative configuration with CRDs.
+- Fully dynamic configuration.
+- Supports native Kubernetes Ingress resource (both v1 and v1beta1).
+- Supports service discovery through Kubernetes Service.
+- Out-of-the-box node health check support.
+- Supports load balancing based on pods (Upstream nodes).
+- Rich [Plugins](/docs/apisix/plugins/batch-requests/) with [custom Plugin](/docs/apisix/plugin-develop/) support.
```
Comment (tao12345666333) [inline docs/en/latest/getting-started.md:111]: Ditto

File: docs/en/latest/getting-started.md [hunk 5]
```diff
@@ -21,68 +26,109 @@ title: Getting Started
...
 
-If you want to deploy apisix-ingress-controller on Prem, we recommend you to use [Helm](https://helm.sh/). Just a few steps
+## Get involved
 
-## Get Involved to Contribute
+You can contribute to the development of APISIX ingress controller. See [Development guide](./contribute.md) for instructions on setting up the project locally.
 
-First, your supports and cooperations to make this project better are appreciated.
-But before you start, please read [How to Contribute](./contribute.md)
+See the [Contribute to APISIX](/docs/general/contributor-guide/) section for details on the contributing flow.
```
Comment (tao12345666333) [inline docs/en/latest/getting-started.md:117]: Ditto

**Review by tao12345666333 — COMMENTED** [review]
Please correct these links

**Comment by navendu-pottekkat** [comment]
@tao12345666333 These links work! I have been using this reference on APISIX docs. What do you suggest we change this to?

I have tested it by building it locally.

https://user-images.githubusercontent.com/49474499/185200506-83a886d4-550f-4c0f-91e9-35346632fdcb.mov

**Comment by navendu-pottekkat** [comment]
The error in CI is from this PR. But the link checker is naive. These links would work fine in the actual build. It is much better than linking to the APISIX website manually.

**Comment by navendu-pottekkat** [comment]
I think we removed this check from the APISIX repo.

**Comment by tao12345666333** [comment]
> 
> Sorry. The error in CI is indeed from this PR. But the link checker is naive. These links would work fine in the actual build. It is much better than linking to the APISIX website manually.

Yes. The checks in CI are for this repo.
I think we have two ways:

1. Remove the check - so everything works fine in the website

2. Modify the link to be a link with the domain name - this works fine whether viewed via GitHub or the website

**Pull request merged**